packages/@aws-cdk-testing/framework-integ/test/aws-elasticloadbalancingv2/test/integ.nlb.attributes.js.snapshot/aws-cdk-nlb-attributes-integ.template.json

```diff
@@ -403,10 +403,6 @@
       "Key": "load_balancing.cross_zone.enabled",
       "Value": "true"
      },
-     {
-      "Key": "ipv6.deny_all_igw_traffic",
-      "Value": "true"
-     },
```

Comment:
That's a scary change, I'm going to assume it's an unintended side effect. Given the integration was being deployed before your change, I assume the integ stack was being deployed successfully

Reply:
Hehe, well that's the thing, I _could not_ deploy the integration test before my change. The deployment failed with
```txt
Resource handler returned message: "Load balancer attribute key 'ipv6.deny_all_igw_traffic' is not supported on load balancers with IP address type 'ipv4'.
```
That's why I renamed the test to, in effect, remove the old test and replace with a template you can deploy (the name I use also aligns with the existing alb test). This particular integration test was added very recently. I can only think that who ever created it only synthesized the test and didn't try deploying.

Reply:
@badmintoncryer Could we get some clarification on this? See #29521

Reply:
Surely I haven't created a snapshot without deploying it... 
I'll try it later. Please wait for a while.

Reply:
@nmussy @Michae1CC 
I tried to deploy `integ.nlb-attributes.ts` but it cannot. I'm really sorry and I don't know how I could create snapshot files...

I have conducted deployment verification for internal ALBs and NLBs across several patterns.
The results are the same for both ALB and NLB.

|`ipAddressType`|`denyAllIgwTraffic`|deployment result|error message|
|-|-|-|-|
|IPV4|true|failed|Load balancer attribute key 'ipv6.deny_all_igw_traffic' is not supported on load balancers with IP address type 'ipv4'.|
|IPV4|false|failed|Load balancer attribute key 'ipv6.deny_all_igw_traffic' is not supported on load balancers with IP address type 'ipv4'.|
|DUAL_STACK|true|success||
|DUAL_STACK|false|failed|Load balancer attribute key 'ipv6.deny_all_igw_traffic' cannot be modified for load balancers of type 'network'. |

Based on these results, I propose the following amendments:

1. If `denyAllIgwTraffic` is defined, return an error if `ipAddressType` is not `DUAL_STACK`.
2. Revise the existing `integ.nlb-attributes.ts` to make it deployable:
- Remove the `denyAllIgwTraffic` setting.
- Instead, set `denyAllIgwTraffic` in `integ.nlb.dualstack.internal.ts`.

For item 2, it might be better to handle it as a separate issue. 
If so, I will take responsibility for addressing it.

Reply:
Thanks for this @badmintoncryer , I agree with you on both points. I'll address your first point here and leave you to fix the integration test. @nmussy I'll create the additional unit tests once I've finished making these changes.

Reply:
@Michae1CC I will handle the resolution of the integration test. Although I have not dealt with this case before, I believe it would be best to wait for the merge of the integration test modification PR before moving this PR to in review.

@nmussy What are your thoughts on this?

Reply:
I don't see the need or point of doing two PRs to accomplish this. We won't be able to merge the fix without updating the integ and its snapshot, and just merging an update of the snapshot seems pointless, it doesn't resolve any issue for CDK users. I'm not really sure who should take precedence in this case, Kazuho should be responsible for this change, but this is Michael's first PR and he's already started working on it. 

I also don't think it's a high priority bug, given that CloudFormation is already catching this issue, this is just not great DX. I'll let you guys decide but I don't think two PRs is the way to go 👍

Reply:
That's a good point, it seems strange to leave the integration testing broken when we've made a fix for it here. Happy to also fix the integration tests ie, revise the `integ.nlb-attributes.ts` test and add update the `integ.nlb.dualstack.internal.ts` test. How does that sound?

Reply:
@badmintoncryer just saw your new PR. I don't mind either way.

Reply:
I apologize, I just noticed this comment. I will close my PR, could you please incorporate the same content?

Reply:
Sure, I'll update this PR and handle integration tests 👍

Reply:
Thank you very much. I apologize for any inconvenience caused.

Reply:
No problem 😄

Reply:
@Michae1CC My only suggestion is that [the PR title should describe the content of the bug being resolved](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request), rather than just mentioning the resolution of integration test errors.

```md
Titles for feat and fix PRs end up in the change log. Think about what makes most sense for users reading the changelog while writing them.
- feat: describe the feature (not the action of creating the commit or PR, for example, avoid words like "added" or "changed")
- fix: describe the bug (not the solution)
```

Since I am unable to conduct community reviews, it would be best for the rest to be checked by @nmussy .

Reply:
Gotcha, maybe something like `integ test for NLB attributes failing to deploy due to setting denyAllIgwTraffic`?

Reply:
I apologize, I finally realized that this was actually a PR intended to resolve an integration test error. My English skills are limited and I made a significant misunderstanding. My suggestion to split the PR also made no sense. I'm sorry.

Without specifically mentioning the integration test, how about using the title: 
'fix(elbv2): unable to deploy an IPv4 load balancer with `denyAllIgwTraffic` enabled'?
or considering it as the addition of a verification feature,
'feat(elbv2): verification of `denyAllIgwTraffic` for IPv4 Load Balancers'

However, I would appreciate it if you could consider nmussy opinion on this matter as well.

Reply:
No need to apologize! And I agree, this PR's goal is not to fix an integration test, it's to prevent this pattern from being synthesized in the first place

Reply:
Thanks both, updated the title